include/jemalloc/internal/arena_inlines_b.h

```diff
@@ -298,7 +300,9 @@ JEMALLOC_ALWAYS_INLINE void
 arena_dalloc_large(tsdn_t *tsdn, void *ptr, tcache_t *tcache, szind_t szind,
     bool slow_path) {
 	assert (!tsdn_null(tsdn) && tcache != NULL);
-	if (szind < tcache_nhbins_get(tcache)) {
+	if (szind < TCACHE_NBINS_MAX &&
+	    !tcache_bin_disabled(szind, &tcache->bins[szind],
+	    tcache->tcache_slow)) {
```

Comment:
Just realized this isn't correct -- it's possible that an allocation was sampled (i.e. szind < SC_NBINS) but it's doesn't meet the first condition (szind < nbins). We should always check the is_sampled condition first, before checking tcache_eligible. Also we cannot assume tcache_nbins > SC_NBINS. Something like:
```
bool is_sampled = config_prof && (szind < SC_NBINS);
if (unlikely(is_sampled)) {
  arena_dalloc_promoted(tsdn, ptr, tcache, slow_path);
} else {
  if (szind < tcache_nbins_get(tcache) && !tcache_bin_disabled()) {
     tcache_dalloc_large
  } else {
    // large_dalloc path
  }
}
```